### PR TITLE
Flink: Add REST catalog shorthand

### PR DESCRIPTION
**Pull request by Fokko**
Currently I'm using the REST Catalog using a `catalog-impl`:

```sql
CREATE CATALOG my_catalog WITH (
  'type'='iceberg',
  'catalog-impl'='org.apache.iceberg.rest.RESTCatalog',
  'uri'='http://localhost:8181',
  's3.access-key-id'='admin',
  's3.secret-access-key'='password',
  's3.endpoint'='http://localhost:9000'
);
```

But it would be nicer to have the `catalog-type` shorthand:

```sql
CREATE CATALOG my_catalog WITH (
  'type'='iceberg',
  'catalog-type'='rest',
  'uri'='http://localhost:8181',
  's3.access-key-id'='admin',
  's3.secret-access-key'='password',
  's3.endpoint'='http://localhost:9000'
);
```

This PR adds this

### CODE REVIEW

**Review by rdblue — APPROVED** [review]
Looks good to me. Thanks, @Fokko!

File: flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    FileIO io = table.io();
+    Assertions.assertThat(io)
+        .as("FileIO should be a HadoopFileIO")
+        .isInstanceOf(HadoopFileIO.class);
+    HadoopFileIO hadoopIO = (HadoopFileIO) io;
+    Assert.assertEquals("my_value", hadoopIO.conf().get("my_key"));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <T> T javaSerAndDeSer(T object) throws IOException, ClassNotFoundException {
```
Comment (stevenzwu) [inline flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java:104]: nit: maybe just `javaSerdes` is enough as method name?

**Review by stevenzwu — APPROVED** [review]

**Comment by rdblue** [comment]
Looks like there are failing tests.

File: docs/flink-getting-started.md [hunk 2]
```diff
@@ -260,6 +260,24 @@ The following properties can be set if using the Hadoop catalog:
...
+  'type'='iceberg',
+  'catalog-type'='rest',
+  'uri'='https://localhost/'
+);
+```
+
+The following properties can be set if using the REST catalog:
+
+* `uri`: The URL to the REST Catalog (Required)
+* `credential`: A credential to authenticate against the REST catalog (Optional)
```
[the remark this answers is not in the document]
Comment (Fokko) [reply docs/flink-getting-started.md]: I like it, thanks!

File: flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  @Test
+  public void testHadoopCatalogLoader() throws IOException, ClassNotFoundException {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, "file:" + warehouse);
+    CatalogLoader loader = CatalogLoader.hadoop("my_catalog", hiveConf, properties);
+    validateCatalogLoader(loader);
+  }
+
+  @Test
+  public void testHiveCatalogLoader() throws IOException, ClassNotFoundException {
```
[the remark this answers is not in the document]
Comment (Fokko) [reply flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java:77]: I like the 1:1 mapping of classes and their tests

File: flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java (hunk 1 above)
Comment (Fokko) [reply flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/TestCatalogLoader.java:104]: Good one, thanks!

**Comment by Fokko** [comment]
Thanks @stevenzwu, @nastra, and @rdblue for the prompt review!

**Comment by stevenzwu** [comment]
@Fokko when you got time, can you also backport it to 1.14 and 1.15?

**Pull request merged**